model/src/main/kotlin/OrtResult.kt

```diff
@@ -57,6 +57,25 @@ data class OrtResult(
         @JsonInclude(JsonInclude.Include.NON_EMPTY)
         val data: CustomData = emptyMap()
 ) {
+    /**
+     * Return all dependencies of the given [pkg], up to and including a depth of [maxDepth] where counting starts at 0
+     * (for the [Package] itself) and 1 are direct dependencies etc. A value below 0 means to not limit the depth.
+     */
+    @Suppress("UNUSED") // This is intended to be mostly used via scripting.
+    fun collectAllDependencies(pkg: Package, maxLevel: Int = -1): SortedSet<PackageReference> {
```

Comment:
Should this also only take the `Identifier` instead of the whole `Package`?

Reply:
That would indeed be more consistent with my other changes in this PR to remove some "convenience" variants of functions that take a `Package` instead of an `Identifier`, but on the other hand this currently is a tad nicer to be used in scripts. I'm mostly undecided, with a slight preference to keep it as it is.

Reply:
Ok.